ts/components/DevicePairingDialog.tsx

```diff
@@ -0,0 +1,293 @@
+import React from 'react';
+import { QRCode } from 'react-qrcode';
+
+import { SessionModal } from './session/SessionModal';
+import { SessionButton } from './session/SessionButton';
+
+interface Props {
+  i18n: any;
+  onClose: any;
+  pubKeyToUnpair: string | null;
+  pubKey: string | null;
+}
+
+interface State {
+  currentPubKey: string | null;
+  accepted: boolean;
+  isListening: boolean;
+  success: boolean;
+  loading: boolean;
+  view:
+    | 'default'
+    | 'waitingForRequest'
+    | 'requestReceived'
+    | 'requestAccepted'
+    | 'confirmUnpair';
+  pubKeyRequests: Array<any>;
+  data: Array<any>;
+}
+
+export class DevicePairingDialog extends React.Component<Props, State> {
+  constructor(props: any) {
+    super(props);
+
+    this.closeDialog = this.closeDialog.bind(this);
+    this.onKeyUp = this.onKeyUp.bind(this);
+    this.startReceivingRequests = this.startReceivingRequests.bind(this);
+    this.stopReceivingRequests = this.stopReceivingRequests.bind(this);
+    this.getPubkeyName = this.getPubkeyName.bind(this);
+
+    this.state = {
+      currentPubKey: this.props.pubKey,
+      accepted: false,
+      isListening: false,
+      success: false,
+      loading: true,
+      view: 'default',
+      pubKeyRequests: [],
+      data: [],
+    };
+  }
+
+  componentDidMount() {
+    this.getSecondaryDevices();
+  }
```

Comment:
I think if you `tslint` this file it complains about promise not being handled. You can ignore the result of the promise by adding `.ignore()` to it.